test/unit/ServerConfigParams.test.ts

```diff
@@ -1,122 +1,156 @@
 
     before(function () {
-      randomAddress = ethers.Wallet.createRandom().address;
+      originalConfig = config.util.toObject(config) as ServerConfigParams;
     });
 
-    beforeEach(function () {
-      app = {
-        url: 'https://fake.rsk',
-        port: 8080,
-        workdir: '/some/dir',
-      } as AppConfig;
+    afterEach(function () {
+      config.util.extendDeep(config, originalConfig);
+    });
+
+    it('Should pass with default configuration', function () {
+      config.util.extendDeep(config, { app: { disableSponsoredTx: false } });
 
-      contracts = {
-        relayHubAddress: randomAddress,
-      } as ContractsConfig;
+      expect(() => verifyServerConfiguration()).not.to.throw();
     });
 
     it('Should fail if disableSponsoredTx is not properly configured', function () {
-      expect(() =>
-        verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-      ).to.throw(ERROR_DISABLE_SPONSOR_TX_NOT_CONFIGURED);
+      config.util.extendDeep(config, {
+        app: { disableSponsoredTx: undefined },
+      });
+      expect(() => verifyServerConfiguration()).to.throw(
+        'Server configuration error:  "app.disableSponsoredTx" is required'
+      );
     });
 
-    it('Should pass if it is sponsored even if none fee is configured', function () {
-      app.disableSponsoredTx = false;
+    it('Should pass if it is sponsored, even if none fee is configured', function () {
+      config.util.extendDeep(config, { app: { disableSponsoredTx: false } });
 
-      expect(() =>
-        verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-      ).not.to.throw();
+      expect(() => verifyServerConfiguration()).not.to.throw();
     });
+
     describe('When is not sponsored', function () {
       beforeEach(function () {
-        app.disableSponsoredTx = true;
+        config.util.extendDeep(config, { app: { disableSponsoredTx: true } });
       });
 
       it('Should fail if gasFeePercentage is a negative value', function () {
-        app.gasFeePercentage = -0.1;
+        config.util.extendDeep(config, { app: { gasFeePercentage: '-1' } });
 
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).to.throw(ERROR_GAS_FEE_PERCENTAGE_NEGATIVE);
+        expect(() => verifyServerConfiguration()).to.throw(
+          ERROR_GAS_FEE_PERCENTAGE_NEGATIVE
+        );
       });
 
       it('Should fail if fixedUsdFee is a negative value', function () {
-        app.fixedUsdFee = -1;
+        config.util.extendDeep(config, { app: { fixedUsdFee: '-1' } });
 
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).to.throw(ERROR_FIXED_USD_FEE_NEGATIVE);
+        expect(() => verifyServerConfiguration()).to.throw(
+          ERROR_FIXED_USD_FEE_NEGATIVE
+        );
       });
 
       it('Should pass even if all fee parameters are undefined', function () {
```

Comment:
What happens on this scenario? The relayServer works properly, it should sponsor or not sponsor transactions? Should we required at least on charging method?

Reply:
That test is enclosed by a describe dedicated only for sponsored transactions. If the server is configured as sponsored it doesn't care about the fees configuration.

Reply:
understood :)